### ISSUE TITLE
poetry build for source naming convention changed between version 1.2.0 and 1.2.1

### ISSUE THREAD

**Issue by Dr-Irv** [python-poetry/poetry#7314]
- **Poetry version**: 1.2.0 behavior as compare to 1.2.1 and up
- **Python version**:  3.9.13
- **OS version and name**: Windows 10
- **pyproject.toml**: https://github.com/pandas-dev/pandas-stubs/blob/main/pyproject.toml

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [N/A ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

We use poetry to build pandas-stubs.

With poetry version 1.2.0, when we do `poetry build`, we would get the following files:
```text
Building pandas-stubs (1.5.2.230105)
  - Building sdist
  - Built pandas-stubs-1.5.2.230105.tar.gz
  - Building wheel
  - Built pandas_stubs-1.5.2.230105-py3-none-any.whl
```

With poetry version 1.2.1 and above, when we do `poetry build`, we get the following files:
```text
Building pandas-stubs (1.5.2.230105)
  - Building sdist
  - Built pandas_stubs-1.5.2.230105.tar.gz
  - Building wheel
  - Built pandas_stubs-1.5.2.230105-py3-none-any.whl
```
Note the change in the sdist of the source, where it used to be `pandas-stubs`, and now it is `pandas_stubs`.  In other words, it used to include a hyphen, and now it is an underscore.

Is this a documented and intentional change?  For stubs packages, it isn't a great thing!

**Comment by neersighted** [comment]
Yes, this was an expected fix for #5782 that brings Poetry in compliance with [PEP 625](https://peps.python.org/pep-0625/). This was fixed in poetry-core [1.3.0](https://github.com/python-poetry/poetry-core/releases/tag/1.3.0).

However, poetry-core 1.3.0 was not brought in to the Poetry tree before [Poetry 1.2.2](https://github.com/python-poetry/poetry/blame/1.2.2/pyproject.toml#L47) -- I suspect you may have mixed up the versions when you were testing this, unless something truly odd is going on.

**Comment by Dr-Irv** [comment]
> Yes, this was an expected fix for #5782 that brings Poetry in compliance with [PEP 625](https://peps.python.org/pep-0625/). This was fixed in poetry-core [1.3.0](https://github.com/python-poetry/poetry-core/releases/tag/1.3.0).

OK thanks for confirming.  Was that change documented anywhere?

**Comment by neersighted** [comment]
In the poetry-core release notes I linked, which link to the PR that changed it, python-poetry/poetry-core#484. Note that Poetry 1.2.2 predates our new practice of including poetry-core release notes in Poetry's own release notes -- Poetry 1.2.2 requires you to click through to see the release notes.
